flow/scripts/synth_preamble.tcl

```diff
@@ -21,27 +21,31 @@ if {[env_var_exists_and_non_empty VERILOG_INCLUDE_DIRS]} {
   set vIdirsArgs [join $vIdirsArgs]
 }
 
-
-# Read verilog files
-foreach file $::env(VERILOG_FILES) {
-  if {[file extension $file] == ".rtlil"} {
-    read_rtlil $file
-  } elseif {[file extension $file] == ".json"} {
-    read_json $file
-  } else {
-    read_verilog -defer -sv {*}$vIdirsArgs $file
-  }
-}
-
-
-
-
 # Read standard cells and macros as blackbox inputs
 # These libs have their dont_use properties set accordingly
 read_liberty -overwrite -setattr liberty_cell -lib {*}$::env(DONT_USE_LIBS)
 read_liberty -overwrite -setattr liberty_cell \
   -unit_delay -wb -ignore_miss_func -ignore_buses {*}$::env(DONT_USE_LIBS)
 
+# Read verilog files
+if {[env_var_exists_and_non_empty SYNTH_USE_SLANG] && !([file extension $::env(VERILOG_FILES)] == ".rtlil")} {
```

Comment:
can there be multiple .rtlil files?

Reply:
I'm not sure anyone is using it but VERILOG_FILES="a.rtil b.rtlil" does work even with this change, I've tested just now

Reply:
Not that anyone will try :-)

VERILOG_FILES="a.v b.rtlil"

?

Reply:
Right, that will stop working if you do it together with SYNTH_USE_SLANG=1, but maybe the fix here is to stop using VERILOG_SOURCES for non-Verilog/SV files

Reply:
RTLIL_FILES, SV_FILES, VERILOG_FILES?

Reply:
I think it's enough to differentiate RTL source passed to a language frontend (`.v`, `.sv`), and Yosys checkpoints (`.rtlil`, `.json`). In the name of minimal breakage for users we can keep `VERILOG_FILES` for the former, and perhaps use `RTLIL_FILE` for the latter.

If we do this split I expect either the control flow in the synthesis script will be such that `VERILOG_FILES` will be ignored whenever `RTLIL_FILE` exists, or we need to explicitly unset `VERILOG_FILES` when invoking synth.tcl from the Makefile. Otherwise the RTL sources will be re-read.

This means it's not possible to input a design partly as .rtlil and partly as .v/.sv, which started this discussion, but I think that's fine.